packages/http/src/client.ts

```diff
@@ -0,0 +1,141 @@
+import { IPrismOutput } from '@stoplight/prism-core';
+import { IHttpOperation } from '@stoplight/types';
+// @ts-ignore
+import logger from 'abstract-logging';
+import { getOrElse } from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { getStatusText } from 'http-status-codes';
+import { defaults } from 'lodash';
+import { parse as parseQueryString } from 'querystring';
+import { parse as parseUrl } from 'url';
+import { createInstance } from '.';
+import { forwarder } from './forwarder';
+import getHttpOperations, { getHttpOperationsFromResource } from './getHttpOperations';
+import { mocker } from './mocker';
+import { router } from './router';
+import { IHttpConfig, IHttpRequest, IHttpResponse, IHttpUrl } from './types';
+import { validator } from './validator';
+
+async function createClientFromResource(specFile: string, defaultConfig: IHttpConfig): Promise<PrismHttp> {
```

Comment:
`createClientFromResource` and `createClientFromString` are almost identical, what about some partial application ?
```typescript
async function createClientFrom(getResource: Function, spec: string, defaultConfig: IHttpConfig): Promise<PrismHttp> {
  const resources = await getResource(spec);
  return createClientFromOperations(resources, defaultConfig);
}
```
```
createClientFrom.bind({}, getHttpOperationsFromResource)
createClientFrom.bind({}, getHttpOperations)
```

Reply:
Yeah, I guess that will work, good point!

Reply:
81d5630